src/snowflake/cli/api/project/schemas/project_definition.py

```diff
@@ -25,3 +29,26 @@ class ProjectDefinition(UpdatableModel):
     streamlit: Optional[Streamlit] = Field(
         title="Streamlit definitions for the project", default=None
     )
+    env: Optional[Dict] = Field(
+        title="Environment specification for this project.",
+        default=None,
+        validation_alias="env",
+    )
+
+    @field_validator("env")
+    @classmethod
+    def _convert_env(cls, env: Optional[Dict]) -> DictWithEnvironFallback:
+        return DictWithEnvironFallback(env if env else {})
```

Comment:
Yes. as suggested in other comment it would be good to move the expansion/rendering step from sql to project definition. So any use of `cli_context.project_definition` would be accessing expanded env/definition. But I decided to clean this up as a follow-up

Reply:
makes sense. thx